From c20222db6bdd36b84241d1acc217a07433745729 Mon Sep 17 00:00:00 2001
From: Yu Yue <yuyue@mprc.pku.edu.cn>
Date: Mon, 31 Oct 2011 17:58:46 +0800
Subject: [PATCH 053/641] Add some functions about arch irq

Fixed error:
noinitramfs.c:(.init.text+0x370): undefined reference to `arch_local_irq_restore`
include/asm-generic/cmpxchg-local.h:24: error: implicit declaration of function 'arch_local_irq_save'

Signed-off-by: Yu Yue <yuyue@mprc.pku.edu.cn>
---
 arch/unicore64/include/asm/irqflags.h |   27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)
 create mode 100644 arch/unicore64/include/asm/irqflags.h

diff --git a/arch/unicore64/include/asm/irqflags.h b/arch/unicore64/include/asm/irqflags.h
new file mode 100644
index 0000000..1a48d4c
--- /dev/null
+++ b/arch/unicore64/include/asm/irqflags.h
@@ -0,0 +1,27 @@
+#ifndef __ASM_UNICORE64_IRQFLAGS_H__
+#define __ASM_UNICORE64_IRQFLAGS_H__
+
+#ifdef __KERNEL__
+
+/**
+ * arch_local_save_flags() - Save the current interrupt enable state.
+ */
+static inline unsigned long arch_local_save_flags(void)
+{
+	/* FIXME */
+	return 0;
+}
+
+/**
+ * arch_local_irq_restore() - restore saved IRQ state
+ * @flags:
+ */
+static inline void arch_local_irq_restore(unsigned long flags)
+{
+	/* FIXME */
+}
+
+#include <asm-generic/irqflags.h>
+
+#endif /* __KERNEL__ */
+#endif /* __ASM_UNICORE64_IRQFLAGS_H__ */
-- 
1.7.9.5

